### ISSUE TITLE
Grid isn't updated after changing the scale factor

### ISSUE THREAD

**Issue by Picalines** [neovide/neovide#2901]
**Describe the bug**

After you change the `neovide_scale_factor`, `lines` and `columns` don't update sometimes, leading to bottom text clipping outside the window

https://github.com/user-attachments/assets/fe6dcb1e-3a9e-40c9-9291-22a52962c358

^ Notice how the statusline becomes invisible

https://github.com/user-attachments/assets/becc9288-2bc2-45f1-9fdb-404980d98f10

^ Grid gets updated if you resize the window manually

**To Reproduce**

Make a mapping/user_command/etc that changes the `neovide_scale_factor`:

```lua
local function zoom(delta)
  return function()
    vim.g.neovide_scale_factor = vim.g.neovide_scale_factor + delta
  end
end

vim.api.nvim_create_user_command('ZoomIn', zoom(0.3), {})
vim.api.nvim_create_user_command('ZoomOut', zoom(-0.3), {})
```

And use the `ZoomIn/ZoomOut` in a keymap

**Expected behavior**
The grid (`lines` and `columns`) should update after the `neovim_scale_factor` changes

Window size (in pixels) shouldn't change

**Desktop (please complete the following information):**
 - macOS 14.7.1
 - Neovide 0.13.3
 - Neovim 0.10.2

**Please run `neovide --log` and paste the contents of the `.log` file created in the current directory here:**

I'll share the logs if necessary - it seems like they contain rendered text info, including folder names and stuff

**Comment by fredizzimo** [comment]
I can't repeat this neither on the latest version nor in 0.13.3.

We did have a bug in older versions of Neovide, but that was fixed here https://github.com/neovide/neovide/pull/2610 and included in Neovim 0.13.2.

This could be a macOS specific, @falcucci, could you take a look

@Picalines, the logs will help in this case, and yes you are correct, the logs contain too much sensitive information, that's something we really should fix at some point. But it should be ok, if you don't open any sensitive files, or otherwise show something on the screen. 

The important lines in the logs are
`Grid matched saved size, skip update.
and `Resizing grid based on window size` based on those we should see if we are trying to send the resize request to Neovim or not.

**Comment by Picalines** [comment]
I've disabled some plugins to reduce the log size and checked for sensitive info: [neovide.log](https://github.com/user-attachments/files/18223777/neovide.log)

In the session i've typed "this is a test buffer" (search for "test buffer"), and then used my keymaps to reproduce the issue. There's a `Resizing grid based on window size` message, but no `Grid matched saved size`

Also it's somewhat strange, but the issue "occurs more often" when i set the `neovide_scale_factor` in a keymap, and "less ofter" when i try to set it manually from the cmdline. I've tried using the `vim.schedule`, but it doesn't fix the issue. Maybe it's just random
